### DISCUSSION TITLE
Run a function every time I visit a specific route and make sure it is running when the DOM is completely read.

### DISCUSSION THREAD

**Discussion by shaheer-haider** [nuxt/nuxt#17740] [Questions]
I am using GSAP with Nuxt 3, I'm trying to run animation with **onMounted** hook, if I try to access DOM elements without GSAP or with GSAP it works on reload but doesn't work when I come from another route. **onMounted** function executes but after coming from another route onMounted runs before the DOM is ready.

Please check the code and let me know if there's anything I can do in Nuxt 3. In Vue, there are solutions to it but I don't know if it is possible to run a function after the DOM is ready after coming from another route.
![Untitled](https://user-images.githubusercontent.com/56408067/162563488-c8b363bb-c1e8-4a92-a49a-9ec752b35d54.jpeg)

**Comment by atinux** [comment]
It would help a lot if you could provide a demo/playground.

**Comment by fabianbeier** [comment]
Hi just had the same issue. Posted an answer/workaround in another Diskussion: nuxt/nuxt#16730 
There's also a sandbox. If you comment out the **definePageMeta** you can see how onMount does not fire the GSAP animation on route-change. @Atinux Any ideas why this happen?
